async.zsh

```diff
@@ -144,19 +144,37 @@ _async_worker() {
 			exit 0
 		}
 
-		# Check for non-job commands sent to worker
-		case $request in
-			_unset_trap) notify_parent=0; continue;;
-			_killjobs)   killjobs; continue;;
-		esac
-
 		# Parse the request using shell parsing (z) to allow commands
 		# to be parsed from single strings and multi-args alike.
 		cmd=("${(z)request}")
 
 		# Name of the job (first argument).
 		local job=$cmd[1]
 
+		# Check for non-job commands sent to worker
+		case $job in
+			_unset_trap) notify_parent=0; continue;;
+			_killjobs)   killjobs; continue;;
+
+
+			# Inherit the parents pid
+			# I can think of two options.
+			# 1: Sending the dir as a parameter to the request.
+			# 2: Using a POSIX equivalent of readlink or a more specialized
+			# tool like pwdx to determine the parents pwd based on
+			# $parent_pid
+			#
+			# Probably 1 is more performant and portable, but I think 2
+			# is more reliable.
+			# Here is an implementation of option 1:
+			# _inheritcwd) cd "`readlink -e /proc/$parent_pid/cwd`"; continue;;
+			#
+			# Below is option 2, which also pushed me to move this
+			# case statement below the cmd and job assignments.
+
+			_inheritcwd) cd ${(Q)cmd[2]}; continue;;
```

Comment:
I used (Q) here because the variable had an extra set of quotes around it when it came through.  Instead of `/home/lattis/dir\ with\ spaces/` it was `'/home/lattis/dir\ with\ spaces/'` which interpreted the `\`s literally.  This is probably no longer needed with the changes above.

I will take a look at that, but I have been using this branch for awhile in my pure_prompt and everything has been working as expected.

Reply:
Huh, I still needed to use `(Q)` for the same reasons stated above, even when there were no quotes around `$PWD` in the `async_job` call